lib/dialects/mssql/index.js

```diff
@@ -213,8 +213,12 @@ Object.assign(Client_MSSQL.prototype, {
   // Get a raw connection, called by the `pool` whenever a new
   // connection needs to be added to the pool.
   acquireRawConnection() {
-    return new Bluebird((resolver, rejecter) => {
-      const settings = Object.assign({}, this.connectionSettings);
+    const client = this;
+    return new Bluebird(async (resolver, rejecter) => {
+      const settings = await client
```

Comment:
Is there a reason why we might want to set connectionConfigProvider each time for a new connection anew and not just set it once on a client when it is created?

Reply:
in `ConnectionConfigProvider` class I processed `config.connection` resolve the connection settings for only one time.  So calling 
```
const settings = await client
        .connectionConfigProvider()
        .resolveConnectionConfig();
```
 will return the processed object direclty without processing anything anymore.

Reply:
Will it?

```
  connectionConfigProvider() {
    return new ConnectionConfigProvider(this);
  },
```

Reply:
any recommendation?

Reply:
As I've said, you can call `connectionConfigProvider ` once after client is created and configuration is known. And probably it should be renamed to `initConnectionConfigProvider` to emphasise it mutates the state.